### ISSUE TITLE
[Ansible] CoreOS pypy libtinfo.so.5 not found

### ISSUE THREAD

**Issue by kgelinas** [kubernetes/contrib#1675]
CoreOS 1153 VMWare image.
Made clean install twice

ansible-playbook --version
ansible-playbook 2.1.1.0
  config file = /home/gelke01/contrib/ansible/scripts/ansible.cfg
  configured module search path = Default w/o overrides

```
TASK [setup] *******************************************************************
<10.0.0.50> ESTABLISH SSH CONNECTION FOR USER: core
<10.0.0.50> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 -o ControlPath=/home/gelke01/.ansible/cp/ansible-ssh-%h-%p-%r 10.0.0.50 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568 `" && echo ansible-tmp-1472733511.21-153745518274568="` echo $HOME/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568 `" ) && sleep 0'"'"''
<10.0.0.50> PUT /tmp/tmpZQSEJQ TO /home/core/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568/setup
<10.0.0.50> SSH: EXEC sftp -b - -C -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 -o ControlPath=/home/gelke01/.ansible/cp/ansible-ssh-%h-%p-%r '[10.0.0.50]'
<10.0.0.50> ESTABLISH SSH CONNECTION FOR USER: core
<10.0.0.50> SSH: EXEC ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=core -o ConnectTimeout=10 -o ControlPath=/home/gelke01/.ansible/cp/ansible-ssh-%h-%p-%r -tt 10.0.0.50 '/bin/sh -c '"'"'sudo -H -S -n -u root /bin/sh -c '"'"'"'"'"'"'"'"'echo BECOME-SUCCESS-zzpeleiojpelxrzvpvcjdforqqoqhyie; LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 PATH=/opt/bin:$PATH python /home/core/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568/setup; rm -rf "/home/core/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568/" > /dev/null 2>&1'"'"'"'"'"'"'"'"' && sleep 0'"'"''
fatal: [IAmCoreMaster1]: FAILED! => {"changed": false, "failed": true, "invocation": {"module_name": "setup"}, "module_stderr": "", "module_stdout": "/bin/bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\r\n/opt/bin/pypy/bin/pypy: /lib64/libssl.so.1.0.0: no version information available (required by /opt/bin/pypy/bin/pypy)\r\n/opt/bin/pypy/bin/pypy: /lib64/libcrypto.so.1.0.0: no version information available (required by /opt/bin/pypy/bin/pypy)\r\n/bin/sh: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)\r\nTraceback (most recent call last):\r\n  File \"app_main.py\", line 75, in run_toplevel\r\n  File \"/home/core/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568/setup\", line 120, in <module>\r\n    exitcode = invoke_module(module, zipped_mod, ZIPLOADER_PARAMS)\r\n  File \"/home/core/.ansible/tmp/ansible-tmp-1472733511.21-153745518274568/setup\", line 28, in invoke_module\r\n    p = subprocess.Popen(['PATH=/opt/bin:$PATH python', module], env=os.environ, shell=False, stdout=subprocess.PIPE, stderr=subprocess.PIPE, stdin=subprocess.PIPE)\r\n  File \"/opt/bin/pypy/lib-python/2.7/subprocess.py\", line 710, in __init__\r\n    errread, errwrite)\r\n  File \"/opt/bin/pypy/lib-python/2.7/subprocess.py\", line 1331, in _execute_child\r\n    raise child_exception\r\nOSError: [Errno 2] No such file or directory\r\n", "msg": "MODULE FAILURE", "parsed": false}
```

```
ldd /opt/bin/pypy/bin/pypy
/opt/bin/pypy/bin/pypy: /lib64/libssl.so.1.0.0: no version information available (required by /opt/bin/pypy/bin/pypy)
/opt/bin/pypy/bin/pypy: /lib64/libcrypto.so.1.0.0: no version information available (required by /opt/bin/pypy/bin/pypy)
        linux-vdso.so.1 (0x00007ffd375eb000)
        libdl.so.2 => /lib64/libdl.so.2 (0x00007fb891a6a000)
        libm.so.6 => /lib64/libm.so.6 (0x00007fb891766000)
        libz.so.1 => /lib64/libz.so.1 (0x00007fb89154e000)
        libssl.so.1.0.0 => /lib64/libssl.so.1.0.0 (0x00007fb8912ce000)
        libcrypto.so.1.0.0 => /lib64/libcrypto.so.1.0.0 (0x00007fb890e4d000)
        libffi.so.6 => /lib64/libffi.so.6 (0x00007fb890c44000)
        libexpat.so.1 => /lib64/libexpat.so.1 (0x00007fb890a16000)
        librt.so.1 => /lib64/librt.so.1 (0x00007fb89080e000)
        libbz2.so.1.0 => /lib64/libbz2.so.1.0 (0x00007fb8905fd000)
        libcrypt.so.1 => /lib64/libcrypt.so.1 (0x00007fb8903c5000)
        libutil.so.1 => /lib64/libutil.so.1 (0x00007fb8901c2000)
        libtinfo.so.5 => not found
        libpthread.so.0 => /lib64/libpthread.so.0 (0x00007fb88ffa5000)
        libc.so.6 => /lib64/libc.so.6 (0x00007fb88fbf5000)
        /lib64/ld-linux-x86-64.so.2 (0x00007fb891c6e000)

```

**Comment by kgelinas** [comment]
same issues with build 1122.1.0

**Comment by ingvagabund** [comment]
@stephenrlouie

**Comment by dshmelev** [comment]
confirm the problem on 1068.10.0

**Comment by dshmelev** [comment]
Works ok on ansible 1.9.6

**Comment by ingvagabund** [comment]
What is the difference between libraries ansible 1.9.6 and 2.1.1.0 brings into the system?

**Comment by tedder** [comment]
I think this is external to Ansible, because pypy itself barfs:

    $ pypy/bin/pypy 
    pypy/bin/pypy: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory

**Comment by ravibhure** [comment]
Its fixed for me, need to link ncurses native path
```
cd /opt/bin/pypy/bin 
sudo ln -s /lib64/libncurses.so.5.9 libtinfo.so.5
```

**Comment by zelahi** [comment]
Has this been fixed in a later version.  I'm still seeing this issue with pip on a coreOS box when I installed pip via Ansible.  

```./pypy --version
./pypy: error while loading shared libraries: libtinfo.so.5: cannot open shared object file: No such file or directory```

I tried the solution @ravibhure mentioned above but it still didn't work for me

**Comment by aerostitch** [comment]
Sorry to comment on a closed issue but as this workaround don't work anymore (as coreos now ships with libncurses 6), I thought I'd put a note here to save other people time.
I created https://bitbucket.org/pypy/pypy/issues/2856/provide-a-version-that-requires-libncurses to ask them to provide a build using libncurses 6.
In the meantime, the workaround I found was using https://bitbucket.org/sqeaky/portable-pypy/downloads that comes with the libs and is compiled using libncurses 6.
